third_party/maya/lib/usdMaya/MayaMeshWriter.cpp

```diff
@@ -38,6 +38,49 @@ PXR_NAMESPACE_OPEN_SCOPE
 
 PXRUSDMAYA_REGISTER_ADAPTOR_SCHEMA(MFn::kMesh, UsdGeomMesh);
 
+namespace {
+
+void _exportReferenceMesh(UsdGeomMesh& primSchema, MObject obj) {
+    MStatus status = MS::kSuccess;
+    MFnDependencyNode dNode(obj, &status);
+    if (!status) { return; }
+
+    auto referencePlug = dNode.findPlug("referenceObject", &status);
+    if (!status || referencePlug.isNull()) { return; }
+
+    MPlugArray conns;
+    referencePlug.connectedTo(conns, true, false);
+    if (conns.length() == 0) { return; }
+
+    auto referenceObject = conns[0].node();
+    if (!referenceObject.hasFn(MFn::kMesh)) { return; }
+
+    MFnMesh referenceMesh(referenceObject, &status);
+    if (!status) { return; }
+
+    const auto* mayaRawPoints = referenceMesh.getRawPoints(&status);
+    const auto numVertices = referenceMesh.numVertices();
+    VtArray<GfVec3f> points(numVertices);
+    for (auto i = decltype(numVertices){0}; i < numVertices; ++i) {
+        const auto floatIndex = i * 3;
+        points[i].Set(mayaRawPoints[floatIndex],
+                        mayaRawPoints[floatIndex + 1],
+                        mayaRawPoints[floatIndex + 2]);
+    }
+
+    static const TfToken _prefToken("Pref");
```

Comment:
We generally don't capitalize attribute names, so we'd prefer "pref" instead of "Pref" unless there's a strong reason to keep it as-is?

Reply:
Pref is the name Arnold uses. I'll add an environment variable that let's the users to configure the Pref name and use that in usdUtils.

Reply:
After discussing the usefulness of the environment variable internally, we decided to skip it. We are going to investigate what's the best way to handle this on the importer's side and create a pr for our solution later on.